Submissions/Laguerta_Week3.playground/Contents.swift

```diff
@@ -0,0 +1,217 @@
+class NamedShape {
+    var numberOfSides: Int = 0
+    var name: String
+    
+    init(name: String) {
+        self.name = name
+    }
+    
+    func simpleDescription() -> String {
+        return "A shape with \(numberOfSides) sides."
+    }
+}
+
+class Circle: NamedShape {
+    var radius: Float = 3.0
+    
+    init(radius: Float, name: String){
+        self.radius = radius
+        super.init(name: name)
+    }
+    
+    func area() -> Float {
+        
+        return radius * radius * 3.14
+    }
+    
+    override func simpleDescription() -> String {
+        return "A shape with no sides."
+    }
+}
+
+
+//: ## Enumerations and Structures
+//: 6.)
+//:
+//: Write a function that compares two `Rank` values by comparing their raw values.
+//:
+enum Rank: Int, CaseIterable {
+    case ace = 1
+    case two, three, four, five, six, seven, eight, nine, ten
+    case jack, queen, king
+    
+    func simpleDescription() -> String {
+        switch self {
+        case .ace:
+            return "ace"
+        case .jack:
+            return "jack"
+        case .queen:
+            return "queen"
+        case .king:
+            return "king"
+        default:
+            return String(self.rawValue)
+        }
+    }
+
+}
+let ace = Rank.ace
+let aceRawValue = ace.rawValue
+
+func compareRank(rank1: Rank, rank2: Rank) -> String {
+    if rank1.rawValue == rank2.rawValue {
+        return "These cards have the same rank."
+    } else if rank1.rawValue > rank2.rawValue {
+        return "\(rank1) is greater than \(rank2)."
+    } else {
+        return "\(rank1) is less than \(rank2)."
+    }
+}
+print(compareRank(rank1: Rank.ace, rank2: Rank.eight))
+print(compareRank(rank1: Rank.king, rank2: Rank.ten))
+
+//: 7.)
+//:
+//: Add a `color()` method to `Suit` that returns “black” for spades and clubs, and returns “red” for hearts and diamonds.
+//:
+enum Suit: CaseIterable {
+    case spades, hearts, diamonds, clubs
+    
+    func simpleDescription() -> String {
+        switch self {
+        case .spades:
+            return "spades"
+        case .hearts:
+            return "hearts"
+        case .diamonds:
+            return "diamonds"
+        case .clubs:
+            return "clubs"
+        }
+    }
+    
+    func color() -> String {
+        switch self {
+        case .spades, .clubs:
+            return "black"
+        case .hearts, .diamonds:
+            return "red"
+        }
+    }
+}
+let hearts = Suit.hearts
+let heartsDescription = hearts.simpleDescription()
+let heartsColor = hearts.color()
+print(Suit.spades.color())
+
+//: 8.)
+//:
+//: Add a third case to `ServerResponse` and to the switch.
+//:
+enum ServerResponse {
+    case result(String, String)
+    case failure(String)
+    case forbidden(String)
+}
+
+let success = ServerResponse.result("6:00 am", "8:09 pm")
+let failure = ServerResponse.failure("Out of cheese.")
+let forbidden = ServerResponse.forbidden("The Restricted Section of the Hogwarts Library")
```

Comment:
This seems too polite for Hogwarts. It needs more life threatening danger to underage wizards.